stockflux-securities-master/src/components/InputForm.js

```diff
@@ -27,47 +37,60 @@ const InputForm = ({ match }) => {
 
   useEffect(() => {
     if (match.params.securityId) {
-      setIsLoading(true);
+      setContextOrState(stateEnum.loading);
       getSecurity(match.params.securityId)
         .then(security => {
-          setIsLoading(false);
-          setErrorMessage(null);
+          setContextOrState(null);
           setSecurityState(security);
         })
         .catch(() => {
-          setIsLoading(false);
-          setErrorMessage("Error, cannot get security");
+          setContextOrState(stateEnum.error);
+          setMessages(["Error, cannot get security"]);
         });
     }
   }, []);
 
   const submitForm = event => {
     event.preventDefault();
-    setInProgress(true);
-
+    setContextOrState(stateEnum.sending);
     const securityObject = {
-      name,
       exchange,
       symbol,
+      name,
       visible,
       enabled
     };
 
-    // call service
-
-    setInProgress(false);
-    console.log(securityObject);
+    postSecurity(securityObject)
+      .then(() => {
+        setMessages(["Security was successfully created"]);
+        setContextOrState(stateEnum.success);
+        setSecurityState({
+          exchange: "",
+          symbol: "",
+          name: "",
+          visible: false,
+          enabled: false
+        });
+        setContextOrState(stateEnum.success);
+      })
+      .catch(err => {
+        if (err instanceof ValidationError) {
+          setMessages(err.messages);
+        } else {
+          setMessages(err.message);
```

Comment:
If you're expecting to pass an array of messages into your alert component, then you need to convert this string to an array. Otherwise, you will get a TypeError when you try to run `.map(...)` on a string.

Reply:
Yes good spot. I guess because if it throws a normal error it will be in the form of as string. 

Changed now.